### PR TITLE
Added docker support for arm64 devices

### LINKED ISSUE #136: Docker image for arm64
Hello, first time ever proposing something on Github. 

**Problem:**
Through experimentation I found that the docker method doesn't work for arm64 (Macbook Air M1 in particular). 

**Solution:**
I found that by changing the first line of the Dockerfile, which is:
"FROM buildpack-deps:bullseye"
to this:
"FROM arm64v8/buildpack-deps:bullseye", the image then supports the arm64 architecture.

After building this local image, it can be used instead of the standard nogil/python that's on dockerhub.

It would be nice to publish this version too on dockerhub, maybe with a name like "nogil/pythonarm" to be more easily used.
Those are my two cents, thanks!

### PR DESCRIPTION
**Pull request by xbit18**
Fix #136 

For simplicity I added a separate Dockerfile for arm64 devices, and added another step to the publish_docker.yml to push the additional image to Dockerhub with tag "nogil-arm64/python"

### CODE REVIEW

File: .github/workflows/publish-docker.yml [hunk 1]
```diff
@@ -29,3 +29,13 @@ jobs:
...
+
+      - name: Build and push Docker image for ARM64 devices
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: docker/DockerfileARM64
+          build-args: |
+            commit=${{ github.sha }}
+          push: true
+          tags: nogil-arm64/python:latest
```
Comment (colesbury) [inline .github/workflows/publish-docker.yml]: I think this needs to be in the `nogil` organization. Like `nogil/python-arm64:latest`

**Review by colesbury — COMMENTED** [review]
Have you tested that this Docker image works on macOS? I see that it's a Debian-based image.

**Comment by xbit18** [comment]
While writing the response to your question I tested again the docker method with the nogil/python image and it appears to work, albeit a warning ("WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested") so I don't know what caused the problems I had previously. 
**Nonetheless** when trying the two different versions there's a clear difference in performance in favour of the arm64 version. I'll now post the answer I was about to send:

@colesbury yes what I did was:
1. clone the nogil repository
2. change the Dockerfile as previously mentioned
3. built the image with 

> docker build . --build-arg commit=af329afc105c9760e1ed17d9de0b6845169e0cbd -t nogil

4. run the example commands listed at this link https://hub.docker.com/r/nogil/python but changing the nogil/python with just nogil, which is my local image
5. And this does indeed work, while if I previously run the commands using the nogil/python image on dockerhub, it gave errors

And yes, I think your suggested name better suits the new image, I will correct it

**Comment by colesbury** [comment]
Thanks!

**Pull request merged**